File: Dsa-day156.java
class Solution {
    public int minDeletionSize(String[] strs) {
        if (strs == null || strs.length == 0) return 0;
        int rows = strs.length;
        int cols = strs[0].length();
        int count = 0;
        
        for (int j = 0; j < cols; j++) {
            for (int i = 1; i < rows; i++) {
                if (strs[i].charAt(j) < strs[i-1].charAt(j)) {
                    count++;
                    break;
                }
            }
        }
        return count;
    }
}

File: Dsa-day156.py
class Solution:
    def minDeletionSize(self, strs: List[str]) -> int:
        if not strs:
            return 0
        rows, cols = len(strs), len(strs[0])
        count = 0
        
        for j in range(cols):
            for i in range(1, rows):
                if strs[i][j] < strs[i-1][j]:
                    count += 1
                    break
        return count